packages/seacas/applications/ejoin/EJoin.C

```diff
@@ -829,7 +831,7 @@ namespace {
                         std::vector<T> &global_values, INT *part_loc_elem_to_global)
   {
     // copy values to master element value information
-    T *local_values = values.data();
+    T *local_values = Data(values);
```

Comment:
I'm interested in this idiom of Data(v) instead of v.data(), but not sure if we should adopt it in other code bases.
I understand that v.data() is not guaranteed to be nullptr if v is empty... but I feel more inclined to ensure we don't make decisions based on the ptr value. In the case that v is not empty, you need the size separately to do anything anyway... I don't know, there are pros/cons, but I'm fine with it.

Reply:
A lot of the vector.data() uses are to interface with the exodus C API.  In that case, the sizes are typically set earlier in defining the metadata and then during the transient loops pointers to the arrays are passed and the exodus library "knows" that if the size is empty, it will be getting a null pointer. If the size is not empty, then the pointer won't be null...  Not sure if that is the best API, but it is what we have...

Reply:
But I agree that good to know how a null pointer will be used and not just blindly pass in either a valid  pointer or a null pointer and hope everything works.  I just need to avoid thinking I'm passing a null pointer and instead passing an invalid pointer to nowhere...